File: baekjoon/js/10828.js
const fs = require("fs");
const buf = fs.readFileSync(0, "utf-8").toString().trim();

const [, ...inputs] = buf.split("\n");

let stack = [];
const commands = {
  push: (x) => {
    stack.push(x);
  },
  pop: () => {
    const val = stack.pop();

    return val ? val : -1;
  },
  size: () => stack.length,
  empty: () => (stack.length === 0 ? 1 : 0),
  top: () => (stack.length === 0 ? -1 : stack[stack.length - 1]),
};

const result = [];

for (const input of inputs) {
  const [command, num] = input.split(" ");

  if (command in commands) {
    const output = commands[command](num);

    if (output) {
      result.push(output);
    }
  }
}

console.log(result.join("\n"));

File: baekjoon/rust/10828.rs
use std::fmt::Write;
use std::io::stdin;

fn main() {
    let mut lines = stdin().lines().skip(1);
    let mut stack = Vec::new();

    let mut output = String::new();

    while let Some(Ok(line)) = lines.next() {
        let mut input = line.split_whitespace();
        let command = input.next().unwrap();

        match command {
            "push" => {
                let x = input.next().unwrap().parse::<i32>().unwrap();

                stack.push(x);
            }
            "pop" => {
                let val = stack.pop();

                let result = if val.is_some() { val.unwrap() } else { -1 };

                writeln!(output, "{}", result).unwrap();
            }
            "size" => {
                writeln!(output, "{}", stack.len()).unwrap();
            }
            "empty" => {
                let result = if stack.is_empty() { 1 } else { 0 };

                writeln!(output, "{}", result).unwrap();
            }
            "top" => {
                let result = if stack.is_empty() {
                    -1
                } else {
                    stack[stack.len() - 1]
                };

                writeln!(output, "{}", result).unwrap();
            }
            _ => (),
        }
    }

    print!("{}", output);
}
